pageserver/src/tenant.rs

```diff
@@ -180,122 +185,76 @@ impl Tenant {
             bail!("Timeline directory already exists, but timeline is missing in repository map. This is a bug.")
         }
```

Comment:
Might as well create the directory here, to be consistent with `branch_timeline`